Mathlib/Data/Quot.lean

```diff
@@ -0,0 +1,709 @@
+/-
+Copyright (c) 2017 Johannes Hölzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl
+-/
+import Mathlib.Init.Data.Quot
+import Mathlib.Logic.Relator
+
+/-!
+# Quotient types
+This module extends the core library's treatment of quotient types (`init.data.quot`).
+## Tags
+quotient
+-/
+
+
+variable {α : Sort _} {β : Sort _}
+
+namespace Setoid
+
+#align setoid.r Setoid.r
+
+theorem ext {α : Sort _} : ∀ {s t : Setoid α},
+    (∀ a b, @Setoid.r α s a b ↔ @Setoid.r α t a b) → s = t
+  | ⟨r, _⟩, ⟨p, _⟩, Eq =>
+  by have : r = p := funext fun a => funext fun b => propext <| Eq a b
+     subst this
+     rfl
+
+end Setoid
+
+namespace Quot
+
+variable {ra : α → α → Prop} {rb : β → β → Prop} {φ : Quot ra → Quot rb → Sort _}
+
+-- mathport name: mk
```

Comment:
You can use `@[inherit_doc]` here to inherit the documentation onto the notation, which is what the linter complains about.